File: access.go
package main

import (
	"flag"
	"fmt"
	"os"

	"golang.org/x/sys/unix"
)

func main() {
	flag.Parse()
	if flag.NArg() != 1 {
		fmt.Printf("usage: access PATH\n")
		os.Exit(1)
	}
	path := flag.Arg(0)
	err := unix.Faccessat(unix.AT_FDCWD, path, unix.F_OK, unix.AT_SYMLINK_NOFOLLOW|1234)
	fmt.Println(err)
}

File: access.c
#include <unistd.h>
#include <fcntl.h>
#include <stdlib.h>
#include <stdio.h>

int main(int argc, char *argv[]) {
	int ret = 0;
	if (argc != 2) {
		printf("usage: access_c PATH\n");
		exit(1);
	}
	faccessat(AT_FDCWD, argv[1], F_OK, AT_SYMLINK_NOFOLLOW);
	perror("");
}
